app/domain/authentication/authn_k8s/copy_text_to_file_in_container.rb

```diff
@@ -42,29 +42,28 @@ def copy_text_to_file_in_container
       #
       # We redirect the output to a log file on the authn-client container
       # that will be written in its logs for supportability.
-      def set_file_content_script(path, content, mode)
+      def bash_script(path, content, mode)
         tmp_cert = "#{path}.tmp"
+        <<~BASH_SCRIPT
+          #!/bin/sh
+          set -e
 
-        "
-#!/bin/sh
-set -e
+          cleanup() {
+            rm -f \"#{tmp_cert}\"
```

Comment:
you can remove the double quotes escaping, throughout all the script, since it is not part of a double-quoted string anymore.
```suggestion
            rm -f "#{tmp_cert}"
```